src/file_pipe_log/pipe.rs

```diff
@@ -177,8 +177,9 @@ impl<F: FileSystem> SinglePipe<F> {
                 }
             } else {
                 self.flush_recycle_metrics(recycle_len);
-                if let Ok(handle) = self.file_system.open(&dst_path) {
-                    return Some(Ok((f.path_id, handle)));
+                match self.file_system.open(&dst_path) {
+                    Ok(handle) => return Some(Ok((f.path_id, handle))),
+                    Err(e) => return Some(Err(Error::Io(e))),
```

Comment:
```suggestion
                Some(self.file_system.open(&dst_path).into())
```
or
```suggestion
                Some(Ok(self.file_system.open(&dst_path)?))
```

Reply:
Nope, the return type of this function `recycle_file` is defined as `Option<Result<(PathID, F::Handle)>>`, so it cannot be adapted to the syntax `?`.

Let me guess how to make it more elegant.

Reply:
Oops...failed. I recommend to use current style of coding.